### PR TITLE
refactor: rm UNASSOCIATED_MODULE_INSTANCE_REFS

### PR DESCRIPTION
**Pull request by SyMind**
## Summary

Pass compilation id in build_module, succeed_module hooks. And remove UNASSOCIATED_MODULE_INSTANCE_REFS for JsModuleWrapper.

## Checklist

- [ ] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

**Comment by SyMind** [comment]
!bench

**Review by jerrykingxyz — APPROVED** [review]

File: crates/rspack_core/src/compiler/make/repair/add.rs [hunk 1]
```diff
@@ -72,6 +74,7 @@ impl Task<MakeTaskContext> for AddTask {
 
     artifact.built_modules.insert(module_identifier);
     Ok(vec![Box::new(BuildTask {
+      compilation_id: self.compilation_id,
```
[the remark this answers is not in the document]
Comment (SyMind) [reply crates/rspack_core/src/compiler/make/repair/add.rs]: yes. I have modified.

**Review by jerrykingxyz — APPROVED** [review]

**Comment by SyMind** [comment]
!eco-ci

**Comment by SyMind** [comment]
!bench

**Pull request merged**